packages/components/src/internal/renderers/AttachmentCard.tsx

```diff
@@ -0,0 +1,138 @@
+                <div
+                    className="attachment-card__body"
+                    onClick={isLoaded ? (_isImage ? _showModal : _onDownload) : undefined}
+                >
+                    <div className="attachment-card__icon">
+                        {_isImage && !isLoaded && <LoadingSpinner msg="" />}
+                        {_isImage && isLoaded && (
+                            <img className={`attachment-card__icon_img ${imageCls}`} src={imageURL} alt={name} />
+                        )}
+                        {!_isImage && <i className={`attachment-card__icon_tile ${iconFontCls}`} />}
+                    </div>
+                    <div className="attachment-card__content">
+                        <div className="attachment-card__name">{name}</div>
+                        <div className="attachment-card__size">
+                            {!isLoaded && <LoadingSpinner msg="Uploading..." />}
+                            {isLoaded && recentlyCreated && (
+                                <>
+                                    <i className="fa fa-check-circle" /> File attached
+                                </>
+                            )}
+                            {isLoaded && !recentlyCreated && size && formatBytes(size)}
+                        </div>
+                    </div>
+                </div>
+                {isLoaded && (
+                    <Dropdown className="attachment-card__menu" componentClass="div" id="attachment-card__menu">
+                        <Dropdown.Toggle useAnchor={true}>
+                            <i className="fa fa-ellipsis-v" />
+                        </Dropdown.Toggle>
+                        <Dropdown.Menu className="pull-right">
+                            <MenuItem onClick={_onDownload}>Download</MenuItem>
+                            {allowRemove && <MenuItem onClick={_onRemove}>Remove attachment</MenuItem>}
+                        </Dropdown.Menu>
+                    </Dropdown>
+                )}
+            </div>
+
+            <Modal bsSize="large" show={showModal} onHide={_hideModal}>
+                <Modal.Header closeButton>
+                    <Modal.Title>
+                        <a onClick={_onDownload} style={{ cursor: 'pointer' }}>
+                            {name}
+                        </a>
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    <img src={imageURL} alt={`${name} image`} title={name} className="attachment-card__img_modal" />
+                </Modal.Body>
+            </Modal>
+        </>
+    );
+});
+
+/**
+ * Formats number of bytes into a human readable string.
+ * Example:
+ * ```
+ * formatBytes(1024);       // 1 KB
+ * formatBytes('1024');     // 1 KB
+ * formatBytes(1234);       // 1.21 KB
+ * formatBytes(1234, 3);    // 1.205 KB
+ * ```
+ * https://stackoverflow.com/a/18650828
+ */
+function formatBytes(bytes: number, decimals = 2): string {
```

Comment:
This seems like it belongs in a utility function file somewhere.

Reply:
Yes, moved to util/utils.ts